search2/python/search/save/handler.py

```diff
@@ -212,21 +196,21 @@ def _process_query_chunk(query_args: dict, search_client_args: dict, field_names
 
     return resp["hits"]["total"]["value"]
 
-
-# TODO: get max_slices from opensearch index settings
-def go(
+async def go( # pylint:disable=invalid-name
     input_body: dict,
     search_conf: dict,
+    publisher: Publisher,
     max_query_size: int = 10_000,
     max_slices=1024,
     keep_alive="1d",
 ):
+    """ Main function for running the query and writing the output """
     if not input_body['compress']:
         print("\nWarning: still compressing outputs\n")
 
     search_client_args = dict(
         hosts=list(search_conf["connection"]["nodes"]),
-        http_compress=True,  # enables gzip compression for request bodies
```

Comment:
Is the comment not actually true?  we could preserve the info in this comment by renaming http_compress -> use_gzip_on_request_body.

Here, as elsewhere, we should probably be using a less flexible datatype than a dict for this config issue.

Reply:
See https://opensearch.org/docs/2.3/clients/python-low-level/